src/main/java/org/nuxeo/ecm/csv/filehandler/CSVImportZipHandler.java

```diff
@@ -0,0 +1,144 @@
+import org.nuxeo.ecm.core.api.ClientException;
+import org.nuxeo.ecm.core.work.api.WorkManager;
+import org.nuxeo.ecm.csv.CSVImporterWork;
+import org.nuxeo.runtime.api.Framework;
+import org.richfaces.model.UploadedFile;
+
+/**
+ * Handles ZIP files to prepare the import.
+ * <p>
+ * The ZIP file will be extracted in nuxeo.csv.blobs.folder, everything in this folder will be deleted first.
+ * <p>
+ * The preparation will if fail if another import is already running.
+ * <p>
+ * The ZIP file must contain a CSV file at the root, the first one found will be used for the import. If there is no CSV
+ * file, this file handler falls back to the default behaviour and returns the uploaded ZIP file.
+ * 
+ * @since 7.3
+ */
+public class CSVImportZipHandler implements CSVImportFileHandler {
+    public CSVImportZipHandler() {
+        super();
+    }
+
+    @Override
+    public boolean accept(UploadedFile uploadedFile, File file) {
+        if (uploadedFile == null) {
+            return false;
+        }
+
+        String csvFileName = FilenameUtils.getName(uploadedFile.getName());
+
+        return "application/zip".equals(uploadedFile.getContentType()) || StringUtils.endsWithIgnoreCase(csvFileName, ".zip");
+    }
+
+    @Override
+    public File handle(File zipFile) throws ClientException {
+        if (zipFile == null) {
+            throw new IllegalArgumentException("zipFile cannot be null");
+        }
+
+        String csvFolderPath = Framework.getProperty(CSVImporterWork.NUXEO_CSV_BLOBS_FOLDER);
+        if (StringUtils.isBlank(csvFolderPath)) {
+            throw new ClientException("The property " + CSVImporterWork.NUXEO_CSV_BLOBS_FOLDER + " is not set");
+        }
+
+        File csvFolder = new File(csvFolderPath);
+        if (!csvFolder.exists()) {
+            csvFolder.mkdirs();
+        }
+        if (!csvFolder.exists() || !csvFolder.isDirectory()) {
+            throw new ClientException("Cannot create directory " + csvFolder.getPath());
+        }
+
+        try {
+            // cleaning of the csv blob folder
+            cleanBlobsFolder(csvFolder);
+
+            // unzip in the csv blob folder
+            unzip(zipFile, csvFolder);
+
+            // csvFile is now the first CSV file in the csv blob folder
+            File csvFile = findCSVFile(csvFolder);
+            if (csvFile != null) {
+                return csvFile;
+            }
+
+            return zipFile;
+        } catch (IOException e) {
+            throw new ClientException("Failed to extract ZIP file " + zipFile.getPath(), e);
+        }
+    }
+
+    protected void unzip(File zipFile, File csvFolder) throws IOException {
+        try (ZipFile zip = new ZipFile(zipFile)) {
+            Enumeration<? extends ZipEntry> entries = zip.entries();
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                File file = new File(csvFolder, entry.getName());
```

Comment:
This is a security hole, as entry.getName() can be manipulated (by manually crafting a special ZIP file) to contain ".." components. You code should have for each entry:

    if (entry.getName().contains("..")) {
        continue;
    }